text/0000-my-feature.md

```diff
@@ -0,0 +1,123 @@
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The `pack detect` command is introduced to the Cloud Native Buildpacks ecosystem, providing a way to run only the detect phase of the buildpack lifecycle. This feature enhances the developer experience by allowing them to quickly determine which buildpacks are relevant for their application without progressing through the entire build process. This was partially discussed in [issue #681](https://github.com/buildpacks/pack/issues/681) but the issue was about implementing a `dry-run` flag. With further discussion with @jjbustamante, decided to go forward with this as a new pack pack command rather than a flag.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+# Motivation
+[motivation]: #motivation
+
+- Simplify and streamline the build process by providing a targeted command for buildpack detection.
+- Reduce build times by skipping unnecessary phases of the buildpack lifecycle.
+- Enable developers to quickly identify which buildpacks are applicable to their application without waiting for the entire build process to complete, or having to `Ctrl+C` midway through.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.
+
+- Define any new terminology.
+- Define the target persona: buildpack author, buildpack user, platform operator, platform implementor, and/or project contributor.
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Ideally, the user should run something like `pack detect --path ./path/to/project --builder builder:name` and it should run the analyze binary, followed by the detect binary in the lifecycle and output the logs / output of it. Preferrably, can also output the `group.toml` to a directory specified with `--detect-output-dir`. The reason to run the analyze binary is to get information about the run image that may impact the outcome of detect via CNB_TARGET_* environment variables.
+
+The following flags should be supported and they will work more or less like `pack build`.
+
+|Short|Long|type|
+|----|-------------|----|
+|-B |--builder|string|
+|-b |--buildpack|strings|
+|-r |--buildpack-registry|string|
+||--detect-output-dir|string|
+|-d|--descriptor|string|
+||--docker-host|string|
+|-e|--env|stringArray|
+||--env-file|stringArray|
+||--extension|strings|
+||--gid|int|
+|-h|--help|
+||--lifecycle-image|string|
+||--network|string|
+|-p|--path|string|
+||--post-buildpack|stringArray|
+||--pre-buildpack|stringArray|
+||--pull-policy|string|
+||--run-image|string|
+||--uid|int|
+||--workspace|string|
+
```

Comment:
What would be the plan for the exit code of this command?

I think we should specify this because it's likely that we'll build tooling around this to run tests. We want the exit code to signal errors and for it to be stable.

Reply:
If I understood your question correctly,
Ideally it should be 0 if a build plan selection is successful. In the case of failure, we can come up with some common cases and define it accordingly.

Reply:
Ok, yes. 0 for success and 1 for failure probably covers most of what is necessary. I'm not sure we need to define a bunch of non-zero exit codes. It does make me wonder what the lifecycle returns 🤔  Maybe we should just make sure the lifecycle exit code is returned from this command? I suspect we'd get good results that way and then this command would be inline with the platform spec.

Reply:
I think we also need to add some examples executions of the command with the expected output, that helps a little bit to understand better what is going to do and if we need to pay attention to some details.